lib/widgets/crowd_league_app.dart

```diff
@@ -122,3 +123,64 @@ class CheckAuth extends StatelessWidget {
         });
   }
 }
+
+class InitializingIndicator extends StatelessWidget {
+  final bool firebaseDone;
+  final bool reduxDone;
+  const InitializingIndicator({
+    @required this.firebaseDone,
+    @required this.reduxDone,
+    Key key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    var message = '';
+    if (!firebaseDone && !reduxDone) {
+      message = 'Waiting for Redux & Firebase...';
+    } else if (!firebaseDone) {
+      message = 'Waiting for Firebase...';
+    } else if (!reduxDone) {
+      message = 'Waiting for Redux...';
+    }
+    return Material(
+      child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            CircularProgressIndicator(),
+            SizedBox(height: 30),
+            Text(message, textDirection: TextDirection.ltr)
+          ]),
+    );
+  }
+}
+
+class ErrorPage extends StatelessWidget {
```

Comment:
This widget just displays the available info if there is an error during intialization.  It's not particularly pretty but it shouldn't ever be seen and if it is, we just need to see the info so I think it's fine.